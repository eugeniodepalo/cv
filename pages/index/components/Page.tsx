import styled from 'styled-components'
import { Component } from 'react'
import { connect } from 'react-redux'
import { Box } from '@rebass/grid'
import { transparentize } from 'polished'
import { StickyContainer, Sticky } from 'react-sticky'
import { Layout, Section, Link } from '~/components'
import { Header, PositionItem, ProjectItem, RepoItem } from './Page/'
import { positions, projects } from '../cv/data'
import { getRepos, getIsFetchingRepos, getReposError } from '../selectors/repos'
import { fetch as fetchRepos } from '../actions/repos'

const Container = styled.div`
  border-top: 6px solid #333;
  background-color: ${(props) => props.theme.bgColor};
`

const Footer = styled.footer`
  text-align: center;
`

const StickyWrapper = styled.div`
  background-color: ${(props) => transparentize(0.1, props.theme.bgColor)};
`

const Error = styled.div`
  display: inline-block;
  color: ${(props) => props.theme.errorColor};
`

const stickyContent = ({ style, isSticky }: any) => (
  <div style={style}>
    <StickyWrapper>
      <Box pb={isSticky ? 0 : 3}>
        <Section>
          <Header isSticky={isSticky} />
        </Section>
      </Box>
    </StickyWrapper>
  </div>
)

const Page = class extends Component<any> {
  public componentDidMount() {
    this.props.fetchRepos()
  }

  public render() {
    return (
      <Layout title="CV">
        <Box mb={3}>
          <Container>
            <StickyContainer>
              <Sticky topOffset={200}>{stickyContent}</Sticky>
              <main>
                <Section title="Positions" id="positions" pt={0}>
                  {positions.map((p) => (
                    <PositionItem position={p} key={p.id} />
                  ))}
                </Section>
                <Section title="Repos" id="repos" pt={0}>
                  {this.props.reposError && <Error>There was an error fetching repos.</Error>}
                  {this.props.isFetchingRepos
                    ? 'Fetching repos...'
                    : (this.props.repos || []).map((r: any) => <RepoItem repo={r} key={r.id} />)}
                </Section>
                <Section title="Projects" id="projects">
                  {projects.map((p) => (
                    <ProjectItem project={p} key={p.id} />
                  ))}
                </Section>
              </main>
            </StickyContainer>
          </Container>
        </Box>
        <Footer>
          Fork this resume on <Link href="https://github.com/eugeniodepalo/cv">GitHub</Link>
        </Footer>
      </Layout>
    )
  }
}

const mapStateToProps = (state: any) => ({
  repos: getRepos(state),
  isFetchingRepos: getIsFetchingRepos(state),
  reposError: getReposError(state)
})

const mapDispatchToProps = { fetchRepos }

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Page)
