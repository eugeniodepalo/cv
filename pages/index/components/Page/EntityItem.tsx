// interface Project {
//   name: string
//   description: string
//   techs: string[]
// }

import styled from 'styled-components'
import { Box, Flex } from '@rebass/grid'
import { Link, BadgeList } from '~/components'

const Title = styled.h1`
  font-size: 1.25rem;
`

export default ({ entity }: any) => (
  <Box as="article" mb={4}>
    <Flex alignItems="center">
      <Box mr={3}>
        <Link href={entity.websiteUrl}>
          <Title>
            <Flex alignItems="center">
              {entity.iconUrl && <Box as="img" mr={1} src={entity.iconUrl}></Box>}
              <Box>{entity.name}</Box>
            </Flex>
          </Title>
        </Link>
      </Box>
      <Box>
        {entity.description}
        <Box mt={1}>
          <BadgeList values={entity.techs} />
        </Box>
      </Box>
    </Flex>
  </Box>
)
