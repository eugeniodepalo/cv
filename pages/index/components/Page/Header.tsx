import styled from 'styled-components'
import { Box, Flex } from '@rebass/grid'
import { Link } from '~/components'
import { info } from '../../cv/data'

const Header = styled.header<any>`
  border-bottom: ${(props) => (!props.isSticky ? `1px dashed ${props.theme.primaryColor};` : '0')};
`

const Picture = styled.img`
  border: 2px dashed ${(props) => props.theme.primaryColor};
`

const Title = styled.h1`
  font-size: 2rem;
`

const Subtitle = styled.h2`
  font-size: 1.5rem;
`

export default ({ isSticky }: any) => (
  <Header isSticky={isSticky}>
    <Flex mb={!isSticky ? 3 : 0}>
      {!isSticky && (
        <Box width={[2 / 12]} mr={4}>
          <Picture src={info.profilePictureUrl} width="100%" />
        </Box>
      )}
      <Box flex="1 1 auto">
        <Flex>
          <Box flex="1 1 auto">
            <Box mb={2}>
              <Title>{info.fullName}</Title>
            </Box>
            <Box mb={isSticky ? 0 : 3}>
              <Subtitle>{info.jobTitle}</Subtitle>
            </Box>
          </Box>
          <Box>
            <Flex alignItems="end">
              <Link href={`mailto:${info.email}`}>Email</Link>
              <Link href={`https://github.com/${info.githubHandle}`} target="_blank">
                GitHub
              </Link>
              <Link href={`https://linkedin.com/in/${info.linkedinHandle}`} target="_blank">
                LinkedIn
              </Link>
            </Flex>
          </Box>
        </Flex>
        {!isSticky && (
          <Box>
            {info.traits.map((t, i) => (
              <p key={i.toString()}>{t}</p>
            ))}
            <p>{info.status}</p>
          </Box>
        )}
      </Box>
    </Flex>
  </Header>
)
