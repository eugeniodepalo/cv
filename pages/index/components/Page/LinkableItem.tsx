import { Box, Flex } from '@rebass/grid'
import { Link, BadgeList } from '~/components'
import { FunctionComponent } from 'react'
import { Project } from '~/cv/data/projects'
import { FetchedRepo } from '~/api'

interface Props {
  entity: Project | FetchedRepo
}

const LinkableItem: FunctionComponent<Props> = ({ entity }) => (
  <Box as="article" mb={4}>
    <Flex alignItems="center">
      <Box mr={3}>
        <Link href={entity.websiteUrl}>
          <h1>
            <Flex alignItems="center">
              {'iconUrl' in entity && entity.iconUrl && <Box as="img" mr={1} src={entity.iconUrl} />}
              <Box>{entity.name}</Box>
            </Flex>
          </h1>
        </Link>
      </Box>
      <Box>
        <div>{entity.description}</div>
        <Box mt={1}>
          <BadgeList values={entity.techs} />
        </Box>
      </Box>
    </Flex>
  </Box>
)

export default LinkableItem
