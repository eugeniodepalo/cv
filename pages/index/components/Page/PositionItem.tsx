import { Position, TeamSize } from '../../cv/data/positions'
import { BadgeList } from '~/components'
import styled from 'styled-components'
import { Fragment } from 'react'
import { Flex, Box } from '@rebass/grid'
import { lighten } from 'polished'
import * as moment from 'moment'

interface Props {
  position: Position
}

const Title = styled.h1`
  font-size: 1.5rem;
`

const DefinitionList = styled.dl`
  display: grid;
  column-gap: 1rem;
  row-gap: 0.25rem;
  grid-template-columns: max-content 1fr;
`

const Label = styled.div`
  background-color: ${(props) => lighten(0.7, props.theme.primaryColor)};
`

const DefinitionItem = ({ label, value, children }: any) => (
  <Fragment>
    <dt>
      <Label>
        <Box p={1}>{label}</Box>
      </Label>
    </dt>
    <dd>{value !== undefined ? value : children}</dd>
  </Fragment>
)

const teamSizeToString = (teamSize: TeamSize) => {
  switch (teamSize) {
    case TeamSize.Small:
      return 'Small (1-3 people)'
    case TeamSize.Medium:
      return 'Medium (3-6 people)'
    case TeamSize.Large:
      return 'Large (7+ people)'
  }
}

export default ({ position }: Props) => {
  const duration = moment.duration(position.endDate.diff(position.startDate)).humanize()

  return (
    <Box as="article" mb={4}>
      <Box mb={3}>
        <Title>{position.name}</Title>
      </Box>
      <Flex flexWrap="wrap">
        <Box width={[1, 1, 4 / 12]} mb={[3, 0]}>
          <DefinitionList>
            <DefinitionItem label="Duration" value={duration} />
            <DefinitionItem label="Role" value={position.role} />
            <DefinitionItem label="Team Size" value={teamSizeToString(position.teamSize)} />
            <DefinitionItem label="Techs">
              <BadgeList values={position.techs} />
            </DefinitionItem>
            <DefinitionItem label="Process">
              <BadgeList values={position.process} />
            </DefinitionItem>
          </DefinitionList>
        </Box>
        <Box width={[1, 1, 8 / 12]}>
          <Box mb={3}>{position.description}</Box>
          <ul>
            {position.highlights.map((highlight, index) => (
              <li key={index.toString()}>
                <Box mb={3}>{highlight}</Box>
              </li>
            ))}
          </ul>
        </Box>
      </Flex>
    </Box>
  )
}
