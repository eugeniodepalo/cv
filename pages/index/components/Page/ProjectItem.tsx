// interface Project {
//   name: string
//   description: string
//   techs: string[]
// }

import styled from 'styled-components'
import { Box, Flex } from '@rebass/grid'
import { Link, BadgeList } from '~/components'

const Title = styled.h1`
  font-size: 1.25rem;
`

export default ({ project }: any) => (
  <Box as="article" mb={4}>
    <Flex alignItems="center">
      <Box mr={3}>
        <Link href={project.websiteUrl}>
          <Title>
            <Flex alignItems="center">
              <Box as="img" mr={1} src={project.iconUrl}></Box>
              <Box>{project.name}</Box>
            </Flex>
          </Title>
        </Link>
      </Box>
      <Box>
        {project.description}
        <Box mt={1}>
          Built with <BadgeList values={project.techs} />
        </Box>
      </Box>
    </Flex>
  </Box>
)
