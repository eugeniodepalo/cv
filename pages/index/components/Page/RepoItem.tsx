export default ({ repo }: any) => <p>{repo.description}</p>
