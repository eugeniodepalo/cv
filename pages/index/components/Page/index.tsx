import Header from './Header'
import PositionItem from './PositionItem'
import EntityItem from './EntityItem'

export { Header, PositionItem, EntityItem }
