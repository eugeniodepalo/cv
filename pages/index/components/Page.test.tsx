import { shallow } from 'enzyme'
import Page from './Page'

describe('With Enzyme', () => {
  it('renders', () => {
    const layout = shallow(<Page />)
    expect(layout.exists()).toBe(true)
  })
})
