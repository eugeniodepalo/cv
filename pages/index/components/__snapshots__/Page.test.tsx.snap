// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering it matches snapshot 1`] = `ReactWrapper {}`;
